styleguide/git.md

```diff
@@ -11,10 +11,9 @@ Joinbox uses a more specific version of the git-flow branching model
 - feature branches may only merged into the develop branch if they are tested and work correctly
 - for each iteration (release cycle, milestone) a new release branch is created from develop as soon everything is ready.
 - the release branch is published to a testing server for acceptance tests by the client. the client should receive the change log of the release branch
-- fixes for the release branch are merged into the release branch
-- the release branch may be merged into develop or another release branch if it is required
-- hot fixes are only created if they must be applied to the master branch for immediate release. they are based on the master branch
-- after a hot fix was released the master branch is merged into the develop branch
+- fixes for the release branch are merged into the release branch. Branches are bases on the HEAD of the release branch
+- the release branch may be merged , after the live release, into develop- and master-branch. If needed also to another release branch.
```

Comment:
i don't see why a merge should not occur earlier than after the release of a release branch. i think this is required in order to get all fixes into all releases. if we dont merge fixes to release rbanches into develop and all other release branches we will have regressions since fixes may skip one or more releases. 

the process needs probably to be like this:
- before a new release branch is created the old release branch(es) needs to be merged back into develop
- all updates to older release branches need to be merged into new release branches in order to keep them up to date
- when a release branch is ready to be released it needs to be merged into the master. on the master branch all internal dependencies (dependencies of the project that are developed inhouse for this specific project) need to be updated in the package.json (after they were versioned by this guide). then the version is updated and the changelog is generated. then the project is tagged. this tag is merged back into the develop and _all_ release branches. the only pain is: how the hell are we going to know which fix landed in whic release (probably by checking in which branch a commit was added first).

Reply:
all updates to older release branches need to be merged into new release branches in order to keep them up to date > may